### PR TITLE
Add drawer menu + New header design 

### PR DESCRIPTION
**Pull request by juliantoledo**
Following work from https://github.com/ampproject/amp-by-example/pull/173 here, to clean up a squashing mistake.

### CODE REVIEW

**Comment by juliantoledo** [comment]
@sebastianbenz I was able to add the flags to expand the accordion and highlight the current example.

I will work on the bug @sriramkrish85 mentioned here:
"letting multiple sidebars open at the same time is a bug and here is how it would be fixed - ampproject/amphtml#3320 "

File: src/20_Components/amp-sidebar.html [hunk 1]
```diff
@@ -56,7 +56,7 @@
 
     `amp-fit-text` expands or shrinks its font size to fit the content within the space given to in the menu.
   -->
-    <amp-sidebar id='sidebar1' layout="nodisplay">
+    <amp-sidebar id='sidebar1' layout="nodisplay" side="right">
```
Comment (sebastianbenz) [inline src/20_Components/amp-sidebar.html]: do we need `id='sidebar1'`?

File: tasks/compile-example.js [hunk 2]
```diff
@@ -84,12 +84,58 @@ module.exports = function(config, updateTimestamp) {
...
+
+  function endStream(cb) {
+    // no examples passed in, no file goes out
+    if (!latestFile || !examples) {
+      cb();
+      return;
+    }
+
+    const stream = this;
+    compileIndex(mapToCategories(examples, null), stream);
```
Comment (sebastianbenz) [inline tasks/compile-example.js]: you can omit the `null` parameter - it is implicitly undefined

File: tasks/compile-example.js [hunk 3]
```diff
@@ -84,12 +84,58 @@ module.exports = function(config, updateTimestamp) {
...
+    compileExamples(examples, stream);
+
+    cb();
+  }
+
+  function compileIndex(categories, stream) {
```
Comment (sebastianbenz) [inline tasks/compile-example.js]: don't pass categories as a parameter. Instead: `categories = mapToCategories(examples)`

File: tasks/compile-example.js [hunk 4]
```diff
@@ -139,61 +181,42 @@ module.exports = function(config, updateTimestamp) {
...
-      // add example to categories instance
-      if (!currentCategory || currentCategory.name != exampleFile.category()) {
-        currentCategory = {
-          name: exampleFile.category(),
-          examples: []
-        };
-        categories.push(currentCategory);
+    sort(examples).filter(exampleFile => !exampleFile.document.metadata.draft)
+    .forEach(function(exampleFile) {
+      if (!exampleFile.document.metadata.draft) {
```
Comment (sebastianbenz) [inline tasks/compile-example.js]: remove ` if (!exampleFile.document.metadata.draft) {` as it is no longer needed and handled by the filter above

File: tasks/compile-example.js [hunk 5]
```diff
@@ -139,61 +181,42 @@ module.exports = function(config, updateTimestamp) {
...
-    gutil.log('Generated ' + indexFile.relative);
-    stream.push(indexFile);
-    cb();
+    });
   }
 
-  function mapToCategories(examples) {
+  function mapToCategories(examples, currentExample) {
     const categories = [];
     let currentCategory;
```
Comment (sebastianbenz) [inline tasks/compile-example.js]: formatting nit: 

```
sort(examples)
  .filter(exampleFile => !exampleFile.document.metadata.draft)
  .forEach(function(exampleFile) {
...
```

File: tasks/compile-example.js [hunk 6]
```diff
@@ -139,61 +181,42 @@ module.exports = function(config, updateTimestamp) {
...
+            name: exampleFile.category(),
+            examples: []
+          };
+          if (currentExample) {
+            currentCategory.selected =
+              (currentCategory.name == currentExample.category());
+          }
+          categories.push(currentCategory);
+        }
+
```
Comment (sebastianbenz) [inline tasks/compile-example.js]: shorter:

```
const selected = currentExample && exampleFile.title() == currentExample.title();
```

File: templates/menu.html [hunk 7]
```diff
@@ -0,0 +1,24 @@
+<amp-sidebar id='sidebar' layout="nodisplay">
```
Comment (sebastianbenz) [inline templates/menu.html]: s/sidebar/menu/ => sidebar is too generic

File: templates/css/styles.css [hunk 8]
```diff
@@ -211,19 +206,135 @@ header h3 {
...
+}
+nav .button > span {
+  display: inline-block;
+  font-weight: 300;
+  line-height: 40px;
+  color: white;  
+  text-transform: uppercase;
+}
+
+/* sidebar menu */
```
Comment (sebastianbenz) [inline templates/css/styles.css]: make all rules specific to sidebar, e.g. start all with #sidebar. 

Also: `#sidebar` is not a good id as it is too generic. Suggestion: `menu`.

File: templates/css/styles.css [hunk 9]
```diff
@@ -211,19 +206,135 @@ header h3 {
...
+  font-size: 14px;
+  height: 48px;
+  padding: 0;
+  color: rgba(0,0,0,.87);
+}
+.selected {
+  font-weight: 700;
+  color: #ff4081;
+}
+.categoryheader {
```
Comment (sebastianbenz) [inline templates/css/styles.css]: use a dash to separate different words => category-header. Also applies to topheader, homelogo, listgrouping,..

File: templates/css/styles.css [hunk 10]
```diff
@@ -211,19 +206,135 @@ header h3 {
...
+}
+
+/* sidebar menu */
+#sidebar {
+  width: 320px;
+}
+.sidebar-close-image {
+   float:right;
+   top: 24px;
+   margin-right: 16px;
```
Comment (sebastianbenz) [inline templates/css/styles.css]: add: `cursor: pointer;`

**Comment by sebastianbenz** [comment]
Nice! Few further comments:

* I find the current text a bit too big, can you use the same font sizes, font weights, line spacing as the MD sample?
* I'd prefer the sidebar background to be white - wdyt?

**Comment by juliantoledo** [comment]
@sebastianbenz, fixed, yes the sidebar background looks better in white.

**Comment by sebastianbenz** [comment]
two Ideas: 

* can we give the expanded accordion header a grey background similar to https://material.google.com/#introduction-goals? 
* Increase line spacing for list items (you're probably doing this already)
* make the touch target for each list item cover the whole sidebar -> don't use a html list but make each link `display: block`:

```
#sidebar .item a {
    color: #333;
    display: block;
    font-weight: 400;
    letter-spacing: 0.1px;
    padding: 10px 24px 10px 46px;
    white-space: nowrap;
}
```

**Comment by juliantoledo** [comment]
Fixed few comments.
I do not think it is possible to give the expanded accordion header a grey background dynamically like in https://material.google.com/#introduction-goals

I also found one issue with amp-sidebar+amp-accordion: https://github.com/ampproject/amphtml/issues/3586

**Comment by sebastianbenz** [comment]
We cannot get the exact same effect, but we can add the background when it is expanded - which should be fine.

**Comment by juliantoledo** [comment]
I added the background to the selected category.

**Comment by sebastianbenz** [comment]
nice!

**Comment by juliantoledo** [comment]
I am staging here: https://abe-stage.firebaseapp.com/ in case you want to test it deeper.

File: tasks/compile-example.js (hunk 2 above)
Comment (juliantoledo) [inline tasks/compile-example.js]: done

File: tasks/compile-example.js (hunk 3 above)
Comment (juliantoledo) [inline tasks/compile-example.js]: done

File: tasks/compile-example.js (hunk 4 above)
Comment (juliantoledo) [inline tasks/compile-example.js]: done

File: tasks/compile-example.js (hunk 6 above)
Comment (juliantoledo) [inline tasks/compile-example.js]: done

File: src/20_Components/amp-sidebar.html (hunk 1 above)
Comment (juliantoledo) [inline src/20_Components/amp-sidebar.html]: we need a different name when we have two in the page, I will change the menu one and leave this one as sidebar.

File: templates/css/styles.css (hunk 8 above)
Comment (juliantoledo) [inline templates/css/styles.css]: done (drawermenu)

File: tasks/compile-example.js (hunk 5 above)
Comment (juliantoledo) [inline tasks/compile-example.js]: done

**Comment by sebastianbenz** [comment]
Idea: add a touch effect to menu items

```
#sidebar .item a:active {
  background-color: #eee;
}
```

**Comment by sebastianbenz** [comment]
The touch target for menu items is a bit small on mobile devices. Can you add the following rule:

```
.amp-mode-touch #drawermenu .item a {
    padding-top: 18px;
    padding-bottom: 17px;
}
```

**Comment by juliantoledo** [comment]
Done @sebastianbenz PTAL

File: templates/menu.html [hunk 11]
```diff
@@ -0,0 +1,29 @@
+<amp-sidebar id='drawermenu' layout="nodisplay">
+  <amp-img class='drawermenu-close-image' src="/img/ic_chevron_left_black_24dp_2x.png" width="32" height="32" alt="Home" on="tap:drawermenu.close" role="button" tabindex="0"></amp-img>
```
Comment (sebastianbenz) [inline templates/menu.html]: can you add `alt="close"

**Comment by juliantoledo** [comment]
done.

**Comment by sebastianbenz** [comment]
* Can you remove logo-icon_1x.png and logo-icon_2x.png

File: templates/css/styles.css [hunk 12]
```diff
@@ -251,19 +249,146 @@ button, .button {
...
+  color: white;  
+  text-transform: uppercase;
+}
+
+/* sidebar menu */
+#drawermenu {
+  width: 320px;
+  background-color: white;
+}
+.drawermenu-close-image {
```
Comment (sebastianbenz) [inline templates/css/styles.css]: use `#drawermenu .close` instead.

File: templates/css/styles.css [hunk 13]
```diff
@@ -251,19 +249,146 @@ button, .button {
...
+}
+#drawermenu .item amp-img {
+    margin-left: 4px;
+    vertical-align: text-bottom;
+}
+#drawermenu .item .selected {
+  font-weight: 700;
+  color: #ff4081;
+}
+#drawermenu .expanded {
```
Comment (sebastianbenz) [inline templates/css/styles.css]: should this be `#drawermenu category .expanded` ?

File: templates/css/styles.css [hunk 14]
```diff
@@ -251,19 +249,146 @@ button, .button {
...
+#drawermenu h4 {
+  font-weight: 700;
+  font-size: 13px;
+  padding: 0;
+  border: 0;
+  background-color: white;
+  width: 100%;
+  line-height: 40px;
+}
+.topheader {
```
Comment (sebastianbenz) [inline templates/css/styles.css]: prefix with `#drawermenu`

File: templates/css/styles.css [hunk 15]
```diff
@@ -251,19 +249,146 @@ button, .button {
...
+  margin-top: 16px;
+  margin-left: 16px;
+}
+.homelogo {
```
Comment (sebastianbenz) [inline templates/css/styles.css]: prefix with `#drawermenu`

File: templates/css/styles.css [hunk 16]
```diff
@@ -251,19 +249,146 @@ button, .button {
...
+  text-align: left;
+  font-weight: 500;
+  font-size: 16px;
+  text-transform: uppercase;
+  line-height: 36px;
+  color: rgba(0,0,0,.87);
+  padding-left: 0px;
+}
+section[expanded] .show-more {
```
Comment (sebastianbenz) [inline templates/css/styles.css]: prefix with `#drawermenu`. This might clash with samples otherwise.

File: templates/css/styles.css [hunk 17]
```diff
@@ -251,19 +249,146 @@ button, .button {
...
+  display: none;
+}
+section:not([expanded]) .show-less {
```
Comment (sebastianbenz) [inline templates/css/styles.css]: prefix with #drawermenu. This might clash with samples otherwise.

File: templates/css/styles.css (hunk 13 above)
Comment (juliantoledo) [inline templates/css/styles.css]: done.

**Comment by juliantoledo** [comment]
Fixed last comments:
https://abe-stage.firebaseapp.com/

**Comment by sebastianbenz** [comment]
LGTM. Great work! The menu works really well!

**Pull request merged**